### ISSUE TITLE
TS7030 "not all code paths return a value" false positive when a never-returning exhaustiveness helper is called (not returned) in a switch default

### ISSUE THREAD

**Issue by valentinmelusson** [microsoft/typescript-go#4680]
## Steps to reproduce
```ts
export function assertUnreachable(x: never): never {
    throw new Error('unreachable: ' + x);
}

export function pick(key: 'a' | 'b' | 'c') {
    switch (key) {
        case 'a':
            return 1;
        case 'b':
            return 2;
        case 'c':
            return 3;
        default:
            assertUnreachable(key);
    }
}
```

`tsconfig.json`:
```
{
    "compilerOptions": {
        "strict": true,
        "noImplicitReturns": true,
        "noEmit": true
    }
}
```

## Behavior with `typescript@6.0`
No errors. assertUnreachable's return type is never, and tsc uses that to treat the default case as control-flow-terminating even without an explicit return — a common exhaustiveness-check idiom.

## Behavior with `tsgo`
error TS7030: Not all code paths return a value.

tsgo doesn't recognize the call to the never-returning function as terminating control flow in this position, so it treats pick as if the default case could fall through without returning.

**Comment by RyanCavanaugh** [comment]
I can't repro this at typescript 7.0.2. Can you confirm what version you're running here?

**Comment by valentinmelusson** [comment]
Closing this as I initially thought this was a behavior difference introduced by TSGO, but after digging further I realized the same behavior already existed in TS6—we were simply ignoring it, so I misread it as a regression on my side.

Sorry for the noise and inconvenience !
